tests/v2/validation/publicapi/clusterroletemplatebindings/crtb_create_steve_test.go

```diff
@@ -0,0 +1,229 @@
+//go:build (validation || infra.any || cluster.any || sanity) && !stress && !extended
+
+package clusterroletemplatebindings
+
+import (
+	"encoding/json"
+	"testing"
+
+	v3 "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
+	management "github.com/rancher/rancher/tests/framework/clients/rancher/generated/management/v3"
+	namegen "github.com/rancher/rancher/tests/framework/pkg/namegenerator"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	"github.com/rancher/rancher/tests/framework/extensions/clusters"
+	"github.com/rancher/rancher/tests/framework/extensions/users"
+	"github.com/rancher/rancher/tests/framework/pkg/session"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type CRTBGenTestSuite struct {
+	suite.Suite
+	client               *rancher.Client
+	session              *session.Session
+	clusterName          string
+	clusterID            string
+	newUser              *management.User
+	crtbJsonTemplateData []byte
+}
+
+func (crtb *CRTBGenTestSuite) TearDownSuite() {
+	crtb.session.Cleanup()
+}
+
+func (crtb *CRTBGenTestSuite) SetupSuite() {
+	testSession := session.NewSession()
+	crtb.session = testSession
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(crtb.T(), err)
+
+	crtb.client = client
+	crtb.clusterName = client.RancherConfig.ClusterName
+	require.NotEmptyf(crtb.T(), crtb.clusterName, "Cluster name to install should be set")
+
+	crtb.clusterID, err = clusters.GetClusterIDByName(crtb.client, crtb.clusterName)
+	require.NoError(crtb.T(), err, "Error getting cluster ID")
+
+	crtb.crtbJsonTemplateData = setupJsonTemplateForClusterRoleTemplateBinding().crtbJsonTemplateData
+	require.NoError(crtb.T(), err)
+
+	crtb.newUser, err = users.CreateUserWithRole(client, users.UserConfig())
+	require.NoError(crtb.T(), err)
+}
+
+func setupJsonTemplateForClusterRoleTemplateBinding() CRTBGenTestSuite {
+	crtbTemplate := v3.ClusterRoleTemplateBinding{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "apiVersion",
+			Kind:       kind,
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: namespace,
+			Name:      name,
+		},
+		ClusterName:      clusterName,
+		RoleTemplateName: roleTemplateName,
+		UserName:         userName,
+	}
+
+	jsonTemplateData, err := json.Marshal(crtbTemplate)
```

Comment:
Thinking more about this and talking to @caliskanugur, we should only use typecast ing as a last resort. We can use the steve client and send in the crtb teamplate directly.
eg: 
```suggestion
	crtbResp, err := steveclient.SteveType("management.cattle.io.clusterroletemplatebindings").Create(crtbTemplate)
```

We wouldn't be needing this function: https://github.com/rancher/rancher/blob/750771d0be06b67e8a72d8630d5883d321a23e53/tests/v2/validation/publicapi/clusterroletemplatebindings/crtb_create_steve_test.go#L79